src/main/java/core/basesyntax/Square.java

```diff
@@ -0,0 +1,30 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class Square extends Figure {
+    private double side;
+    private Random random = new Random();
+
+    public Square(double side, String color) {
+        super(color);
+        this.side = side;
+    }
+
+    public Square() {
+        setColor(random);
+        this.color = getColor();
+        side = random.nextDouble();
+    }
```

Comment:
we don't need this consturctor. Figures don't need to be bothered with randomized, we should strive for separation of concerns - it's FigureSupplier's job. For all figures, it's enough to just have a constructor with all their fields as parameters